packages/astro/src/runtime/client/dev-toolbar/helpers.ts

```diff
@@ -0,0 +1,70 @@
+type NotificationPayload = {
+	state: true;
+	level?: 'error' | 'warn' | 'info';
+} | {
+	state: false
+};
+
+type AppStatePayload = {
+	state: boolean;
+};
+
+type AppToggledEvent = (opts: { state: boolean }) => void;
+
+type ToolbarPlacementUpdatedEvent = (opts: { placement: 'bottom-left' | 'bottom-center' | 'bottom-right' }) => void;
+
+export class ToolbarAppEventTarget extends EventTarget {
+	constructor() {
+		super();
+	}
+
+	toggleNotification(options: NotificationPayload) {
+		this.dispatchEvent(
+			new CustomEvent('toggle-notification', {
+				detail: {
+					state: options.state,
+					level: options.state === true ? options.level : undefined,
+				},
+			})
```

Comment:
Is this object and `NotificationPayload` supposed to be the same? Like, is  the `'toggle-notification'` event supposed to return the same `NotificationPayload` type?

Reply:
Well, not return, but yes the payload for `toggle-notification` is the same as the type